scripts/BUILD.bazel

```diff
@@ -44,18 +44,29 @@ kt_jvm_binary(
     ],
 )
 
-REGEX_PATTERN_CHECK_ASSETS = retrieve_regex_pattern_check_assets("regex_pattern_assets")
+# TODO(#3428): Refactor textproto assets to subpackage level.
+REGEX_PATTERN_CHECK_ASSETS = generate_regex_assets_list_from_text_protos(
+    name = "regex_asset_files",
+    file_content_validation_file_names = [
+        "file_content_validation_checks",
+    ],
+    filepath_pattern_validation_file_names = [
+        "filename_pattern_validation_checks",
+    ],
+)
 
 kt_jvm_library(
     name = "regex_check_assets",
     data = REGEX_PATTERN_CHECK_ASSETS,
-    visibility = [":oppia_script_binary_visibility"],
+    visibility = [":oppia_script_test_visibility"],
 )
 
 kt_jvm_binary(
     name = "regex_pattern_validation_check",
     testonly = True,
     data = REGEX_PATTERN_CHECK_ASSETS,
     main_class = "org.oppia.android.scripts.regex.RegexPatternValidationCheckKt",
-    runtime_deps = ["//scripts/src/java/org/oppia/android/scripts/regex:regex_pattern_validation_check_lib"],
+    runtime_deps = [
+        "//scripts/src/java/org/oppia/android/scripts/regex:regex_pattern_validation_check_lib",
+    ],
```

Comment:
Nit: prefer keeping this on one line if it fits.

Reply:
This doesn't fits in one line, therefore had to do this.